packages/react-core/src/components/SearchInput/SearchInput.tsx

```diff
@@ -30,70 +53,245 @@ export interface SearchInputProps extends Omit<React.HTMLProps<HTMLDivElement>,
-              <span className={css(styles.searchInputCount)}>
-                <Badge isRead>{resultsCount}</Badge>
-              </span>
-            )}
-            {!!onNextClick && !!onPreviousClick && (
-              <span className={css(styles.searchInputNav)}>
-                <Button variant={ButtonVariant.plain} aria-label="Previous" onClick={onPreviousClick}>
-                  <AngleUpIcon />
-                </Button>
-                <Button variant={ButtonVariant.plain} aria-label="Next" onClick={onNextClick}>
-                  <AngleDownIcon />
-                </Button>
-              </span>
+          {value && (
+            <span className={css(styles.searchInputUtilities)}>
+              {resultsCount && (
+                <span className={css(styles.searchInputCount)}>
+                  <Badge isRead>{resultsCount}</Badge>
+                </span>
+              )}
+              {!!onNextClick && !!onPreviousClick && (
+                <span className={css(styles.searchInputNav)}>
+                  <Button variant={ButtonVariant.plain} aria-label="Previous" onClick={onPreviousClick}>
+                    <AngleUpIcon />
+                  </Button>
+                  <Button variant={ButtonVariant.plain} aria-label="Next" onClick={onNextClick}>
+                    <AngleDownIcon />
+                  </Button>
+                </span>
+              )}
+              {!!onClear && (
+                <span className="pf-c-search-input__clear">
+                  <Button variant={ButtonVariant.plain} aria-label={resetButtonLabel} onClick={onClear}>
+                    <TimesIcon />
+                  </Button>
+                </span>
+              )}
+            </span>
+          )}
+        </div>
+        {attributes.length > 0 && (
+          <>
+            <Button
+              className={showSearchMenu && 'pf-m-expanded'}
+              variant={ButtonVariant.control}
+              aria-label={openMenuButtonAriaLabel}
+              onClick={onToggle}
+              aria-expanded={showSearchMenu}
+            >
+              <CaretDownIcon />
+            </Button>
+            {!!onSearch && (
+              <Button variant={ButtonVariant.control} aria-label={submitSearchButtonLabel} onClick={onSearchHandler}>
+                <ArrowRightIcon />
+              </Button>
             )}
-            {!!onClear && (
-              <span className="pf-c-search-input__clear">
-                <Button variant={ButtonVariant.plain} aria-label="Clear" onClick={onClear}>
-                  <TimesIcon />
-                </Button>
-              </span>
-            )}
-          </span>
+          </>
         )}
-      </div>
+      </InputGroup>
+      {attributes.length > 0 && showSearchMenu && (
+        <div className={css(styles.searchInputMenu)}>
+          <Form>
+            {buildFormGroups()}
+            <ActionGroup>
+              <Button variant="primary" onClick={onSearchHandler}>
```

Comment:
Random but should this have `type="submit"`?

```suggestion
              <Button type="submit" variant="primary" onClick={onSearchHandler}>
```